src/lib.rs

```diff
@@ -225,20 +240,92 @@ impl Wiki {
         Ok(())
     }
 
+    /// Add a directory for storing files to the generated html sites or read stored
+    /// files of existing filestorage. Afterwards file links will be added to the
+    /// generated html site.
+    pub fn read_files(&mut self, file_folder: &str, output: &str) -> Result<()> {
+
+        //create the default file path
+        let output_path = PathBuf::from(output);
+        let file_directory = PathBuf::from(output).join(file_folder);
+
+        if !Path::new(file_folder).exists(){
+            info!("Creating directory for file input: {:?}.", file_directory.to_str());
+                   fs::create_dir(&file_directory);
+        }
+
+        for path in &self.input_paths{
+            let mut count = 0;
+
+            //create the path for the html site or just find it
+            let page_files_path = file_directory.join(&path.path).to_str().ok_or_else(||
+                                  "Couldn't create Path to html file!")?.replace(".md","");
+
+            if !PathBuf::from(page_files_path.as_str()).exists(){
+                info!("Creating directory for {}'s files: {}.", path.path.to_str().ok_or_else(|| "Path not found!")?,
+                                                                page_files_path.as_str());
+                fs::create_dir_all(page_files_path.as_str());
+            }
+
+            //add the path
+            let file_path = Path::new(page_files_path.as_str());
+            //read all the files
+            let files = file_path.read_dir()?;
+
+            //now attach the files to the html sites folder
+            for entry in files {
```

Comment:
Here you can get rid of the extra var but write
`for entry in file_path.read_dir()? { ...`